enhancements/two-nodes-openshift/2no.md

```diff
@@ -113,32 +119,57 @@ In both cases, the control-plane's dependence on etcd will cause it to respond w
+User creation of a two-node control-plane is possible via the Assisted Installer and the Agent-Based Installer (ABI). The initial implementation will focus on providing support for the Assisted Installer in managed cluster environments (i.e. ACM), followed by stand-alone cluster support via the Agent-Based Installer.
+The requirement that the cluster can be deployed using only 2 nodes is key because requiring a third baremetal server for installation can be expensive when deploying baremetal at scale. To accomplish this, deployments will use one of the target machines as the bootstrap node before it is rebooted into a control-plane node.
+
+A critical transition during bootstrapping is when the bootstrap reboots into the control plane node. Before this reboot, it needs to be removed from the etcd cluster so that quorum can be maintained as the machine reboots into a second control-plane.
+
+Otherwise, the procedure follows the standard flow except for the configuration of 2 nodes instead of 3.
+
+Because BMC passwords are being collected to initialize fencing, the SaaS offering will not be available (to avoid storing customer BMC credentials in a Red Hat database). ZTP may be considered in the future, but this will need further evaluation to ensure passwords are appropriately handled.
+
+Everything else about cluster creation will be an opaque implementation detail not exposed to the user.
+
+##### Transitioning to a RHEL-HA Controlled Cluster
+Three aspects of cluster creation need to happen for a vanilla two-node cluster to have RHEL-HA functioning as described in the proposal.
+1. Initializing the RHEL-HA cluster
+2. Transitioning control of etcd to RHEL-HA
+3. Enabling fencing in RHEL-HA
 
-Otherwise, the procedure follows the standard flow except for the configuration of 2 nodes instead of 3. At this time we've discussed the collection of RedFish details (including passwords!) for each node. This is needed for the RHEL-HA configuration by leveraging the BareMetalHost CRDs populated from the baremetal platform specification in the install-config. There are open questions on how to ensure that pacemaker is the only entity responsible for fencing to prevent conflicting requests to change the machine state between pacemaker and the baremetal operator. Preventing conflicting fencing logic is also important for optional operators like Node Health Check, Self Node Remediation, and Fence Agents Remediation, but these should not be present during installation.
+We propose the inclusion of a new in-cluster operator responsible for the remediation of 2NO resource agents.
```

Comment:
I'm not sure I understand this part.
- Even though I'm pro on a new operator IIUC the decision was we are trying to embed the functionality in existing operators .
- I'm struggling with the term `remediation` here, I'm not sure it's appropriate in this context. The way I understand it, the operator only sets up the resource agents and the agents are responsible for the fencing/remediation

Reply:
As discussed in our 1/1 this morning, my prediction was that the outcome of today's 2NO sync would be that we'd reestablish the desire for an in-cluster operator to handle the fencing setup.

If the in-cluster operator was responsible for initializing  pacemaker and making sure its resource agents were running healthily, my thought was that this makes the operator's job management of pacemaker resource agents. (i.e. pacemaker remediation, not cluster remediation)

My prediction was wrong. :) we are instead looking at the simple flow of just using MCO to stick a file on disk, so my new draft will remove all references to this operator (outside of the parts that talk about it being an option in case we need it for initializing pacemaker).